packages/m2m-gateway/src/routers/purposeRouter.ts

```diff
@@ -95,7 +95,11 @@ const purposeRouter = (
     .post("/purposes", async (req, res) => {
       const ctx = fromM2MGatewayAppContext(req.ctx, req.headers);
       try {
-        return res.status(501).send();
+        validateAuthorization(ctx, [M2M_ADMIN_ROLE]);
+
+        const purpose = await purposeService.createPurpose(req.body, ctx);
+
+        return res.status(201).send(m2mGatewayApi.Purpose.parse(purpose));
```

Comment:
I just realized that in the two calls to create a certified attribute and a consumer delegation I returned 200 instead of 201 🥲  Could you take advantage of this PR to fix those, and also fix the corresponding API tests?